Extend ONNX FE with `Col2Im` operator

daehyun99 opened the pull request:
### Details:
 - Extend ONNX FE with `Col2Im` operator
 - Unblocked python tests in `src\frontends\onnx\tests\tests_python\test_backend.py`
 - Add test code and prototxt model for testing `Col2Im`

### Issue number(s) that this pull request fixes
- Fixes #30144 

### Tickets:
 - [CVS-99952](https://jira.devtools.intel.com/browse/CVS-99952)

daehyun99 commented:
Hi maintainers, I have completed the task and included the test codes.

Please note that the `IE_GPU` test is showing as failed because I ran the tests in a Colab environment without an Intel GPU.
Please review the code when you have a moment.

Thanks.

- Test Result Log
```sh
Running main() from /content/openvino/src/frontends/tests/frontend/shared/gtest_main_manifest/main.cpp:20
Note: Google Test filter = *col2im*:-:ONNXLoadTest/FrontEndLoadFromTest.testLoadFromTwoFiles/onnx:ONNXLoadTest/FrontEndLoadFromTest.testLoadFromTwoStreams/onnx
[==========] Running 18 tests from 3 test suites.
[----------] Global test environment set-up.
[----------] 6 tests from INTERPRETER
[ RUN      ] INTERPRETER.onnx_col2im_default
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_default (265 ms)
[ RUN      ] INTERPRETER.onnx_col2im_dilations
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_dilations (65 ms)
[ RUN      ] INTERPRETER.onnx_col2im_pads
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_pads (64 ms)
[ RUN      ] INTERPRETER.onnx_col2im_strides
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_strides (65 ms)
[ RUN      ] INTERPRETER.onnx_col2im_batch
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_batch (66 ms)
[ RUN      ] INTERPRETER.onnx_col2im_channel
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_channel (70 ms)
[----------] 6 tests from INTERPRETER (595 ms total)

[----------] 6 tests from IE_GPU
[ RUN      ] IE_GPU.onnx_col2im_default
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:116:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_default (319 ms)
[ RUN      ] IE_GPU.onnx_col2im_dilations
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:116:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_dilations (12 ms)
[ RUN      ] IE_GPU.onnx_col2im_pads
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:116:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_pads (11 ms)
[ RUN      ] IE_GPU.onnx_col2im_strides
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:116:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_strides (11 ms)
[ RUN      ] IE_GPU.onnx_col2im_batch
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:116:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_batch (11 ms)
[ RUN      ] IE_GPU.onnx_col2im_channel
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:116:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_channel (12 ms)
[----------] 6 tests from IE_GPU (376 ms total)

[----------] 6 tests from IE_CPU
[ RUN      ] IE_CPU.onnx_col2im_default
```

daehyun99 commented:
Hmm, Should I apply `clang-format` to the test code?
I think that it reduces code readability.

daehyun99 commented:
Hi, devs
Passed the unit test of `Col2Im`'s ONNX front in Colab envs. (Please note that `IE_GPU` tests were not conducted as I am working in a Colab environment.)

I will solve CI unit test error and `clang-format` error.

Thanks.

<details>

> !/content/openvino/bin/intel64/Debug/ov_onnx_frontend_tests --gtest_filter=*col2im*

```sh
Running main() from /content/openvino/src/frontends/tests/frontend/shared/gtest_main_manifest/main.cpp:20
Note: Google Test filter = *col2im*:-:ONNXLoadTest/FrontEndLoadFromTest.testLoadFromTwoFiles/onnx:ONNXLoadTest/FrontEndLoadFromTest.testLoadFromTwoStreams/onnx
[==========] Running 18 tests from 3 test suites.
[----------] Global test environment set-up.
[----------] 6 tests from INTERPRETER
[ RUN      ] INTERPRETER.onnx_col2im_default
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_default (174 ms)
[ RUN      ] INTERPRETER.onnx_col2im_dilations
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_dilations (48 ms)
[ RUN      ] INTERPRETER.onnx_col2im_pads
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_pads (44 ms)
[ RUN      ] INTERPRETER.onnx_col2im_strides
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_strides (42 ms)
[ RUN      ] INTERPRETER.onnx_col2im_batch
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_batch (43 ms)
[ RUN      ] INTERPRETER.onnx_col2im_channel
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] INTERPRETER.onnx_col2im_channel (43 ms)
[----------] 6 tests from INTERPRETER (395 ms total)

[----------] 6 tests from IE_GPU
[ RUN      ] IE_GPU.onnx_col2im_default
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:110:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_default (21 ms)
[ RUN      ] IE_GPU.onnx_col2im_dilations
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:110:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_dilations (7 ms)
[ RUN      ] IE_GPU.onnx_col2im_pads
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:110:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_pads (6 ms)
[ RUN      ] IE_GPU.onnx_col2im_strides
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:110:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_strides (6 ms)
[ RUN      ] IE_GPU.onnx_col2im_batch
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:110:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_batch (7 ms)
[ RUN      ] IE_GPU.onnx_col2im_channel
unknown file: Failure
C++ exception with description "Exception from src/inference/src/cpp/core.cpp:110:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:287:
[GPU] Context was not initialized for 0 device


" thrown in the test body.
[  FAILED  ] IE_GPU.onnx_col2im_channel (6 ms)
[----------] 6 tests from IE_GPU (53 ms total)

[----------] 6 tests from IE_CPU
[ RUN      ] IE_CPU.onnx_col2im_default
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] IE_CPU.onnx_col2im_default (249 ms)
[ RUN      ] IE_CPU.onnx_col2im_dilations
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] IE_CPU.onnx_col2im_dilations (92 ms)
[ RUN      ] IE_CPU.onnx_col2im_pads
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] IE_CPU.onnx_col2im_pads (94 ms)
[ RUN      ] IE_CPU.onnx_col2im_strides
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] IE_CPU.onnx_col2im_strides (88 ms)
[ RUN      ] IE_CPU.onnx_col2im_batch
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] IE_CPU.onnx_col2im_batch (85 ms)
[ RUN      ] IE_CPU.onnx_col2im_channel
[   INFO   ] Verifying match of <= 22 mantissa bits (24 bits precision - 2 tolerance). 0 value(s) below min_signal: 0 Loosest match found is 24 mantissa bits.
[       OK ] IE_CPU.onnx_col2im_channel (89 ms)
[----------] 6 tests from IE_CPU (698 ms total)

[----------] Global test environment tear-down
[==========] 18 tests from 3 test suites ran. (1146 ms total)
[  PASSED  ] 12 tests.
[  FAILED  ] 6 tests, listed below:
[  FAILED  ] IE_GPU.onnx_col2im_default
[  FAILED  ] IE_GPU.onnx_col2im_dilations
[  FAILED  ] IE_GPU.onnx_col2im_pads
[  FAILED  ] IE_GPU.onnx_col2im_strides
[  FAILED  ] IE_GPU.onnx_col2im_batch
[  FAILED  ] IE_GPU.onnx_col2im_channel

 6 FAILED TESTS
```

</details>

daehyun99 commented on src/frontends/onnx/frontend/src/op/col2im.cpp:
```diff
@@ -0,0 +1,67 @@
+#include "exceptions.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "utils/common.hpp"
+using namespace ov::op::v15;
```
Apply

daehyun99 commented on src/frontends/onnx/tests/onnx_import.in.cpp:
```diff
@@ -320,6 +320,131 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_expand_model_with_initializers) {
+
+    // Expected Output: [2, 1, 3, 3]
+    std::vector<float> single_batch_output = {1.0f, 2.0f, 1.0f, 2.0f, 4.0f, 2.0f, 1.0f, 2.0f, 1.0f};
+    std::vector<float> expected_output;
```
it is not meaningfull.
Because it is test code and testing data's size is small.

daehyun99 commented on src/frontends/onnx/frontend/src/op/col2im.cpp:
```diff
@@ -0,0 +1,67 @@
+            ->outputs()};
+}
+
+ONNX_OP("Col2Im", OPSET_SINCE(1), ai_onnx::opset_18::col2im);
```
It is right.
The ONNX documentation specifies that Col2Im was introduced in opset 18.

However, if we set it to `OPSET_SINCE(18)`, the backend tests fail for lower opsets (e.g., 1, 5, 11, etc.).
Since `18` is technically the first official version for `Col2Im`, setting it to `OPSET_SINCE(1)` ensures broader compatibility with models that might be tagged with lower opset versions.

For reference, `src/frontends/onnx/frontend/src/op/compress.cpp` also uses `OPSET_SINCE(1)` for the same reason.

daehyun99 committed:
Feat: Extend ONNX FE with `Com2Im` operator (#30144)

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

daehyun99 commented:
I am done. just waiting review

mvafin committed:
Merge branch 'master' into ENH/30144-1

daehyun99 commented:
@mvafin 
I believe the cause of the test failure is not related to the changes in this PR.

```sh
Run # Skip test_onnx/test_zoo_models and test_onnx/test_backend due to long execution time - ONNX Model Zoo tests are run separately
```

`The test_onnx/test_backend` test was supposed to be skipped, but it was not, which led to the import error below.
Once this issue is fixed, please rerun the GitHub tests. I believe they will then pass.

Thanks.

```sh
Run # Skip test_onnx/test_zoo_models and test_onnx/test_backend due to long execution time - ONNX Model Zoo tests are run separately
  # Skip test_onnx/test_zoo_models and test_onnx/test_backend due to long execution time - ONNX Model Zoo tests are run separately
  python3 -m pytest -sv ${INSTALL_TEST_DIR}/onnx -k 'not cuda' \
    --junitxml=${INSTALL_TEST_DIR}/TEST-onnx_frontend.graph_iterator_enabled.xml \
    --ignore=${INSTALL_TEST_DIR}/onnx/test_python/test_zoo_models.py
  shell: bash --noprofile --norc -e -o pipefail {0}
  env:
    PIP_CACHE_PATH: /mount/caches/pip/linux
    DEBIAN_FRONTEND: noninteractive
    INSTALL_DIR: /home/runner/_work/openvino/openvino/install
    INSTALL_TEST_DIR: /home/runner/_work/openvino/openvino/install/tests
    PIP_CACHE_DIR: /mount/caches/pip/linux/24.0
    PIP_INSTALL_PATH: /venv/lib/python3.11/site-packages
    PIP_INSTALL_COMMAND: python3 -m pip install
============================= test session starts ==============================
platform linux -- Python 3.11.15, pytest-9.0.3, pluggy-1.6.0 -- /venv/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.11.15', 'Platform': 'Linux-5.15.0-1102-azure-x86_64-with-glibc2.35', 'Packages': {'pytest': '9.0.3', 'pluggy': '1.6.0'}, 'Plugins': {'forked': '1.6.0', 'anyio': '4.13.0', 'metadata': '3.1.1', 'html': '4.2.0', 'xdist': '3.8.0'}, 'CI': 'true'}
rootdir: /__w/openvino/openvino
plugins: forked-1.6.0, anyio-4.13.0, metadata-3.1.1, html-4.2.0, xdist-3.8.0
collecting ... collected 425 items / 1 error

==================================== ERRORS ====================================
____ ERROR collecting install/tests/onnx/tests/tests_python/test_backend.py ____
ImportError while importing test module '/__w/openvino/openvino/install/tests/onnx/tests/tests_python/test_backend.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
install/tests/onnx/tests/tests_python/test_backend.py:9: in <module>
    from tests import (
E   ImportError: cannot import name 'xfail_issue_99952' from 'tests' (/__w/openvino/openvino/install/tests/onnx/tests/__init__.py)
- generated xml file: /__w/openvino/openvino/install/tests/TEST-onnx_frontend.graph_iterator_enabled.xml -
=========================== short test summary info ============================
ERROR install/tests/onnx/tests/tests_python/test_backend.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.73s ===============================
Error: Process completed with exit code 2.
```

mvafin commented:
@daehyun99 The failure is caused by your changes. You removed `xfail_issue_99952` but it is still imported in tests

daehyun99 committed:
Fix: Fix Col2Im test failing

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

daehyun99 committed:
Fix: Fix Col2Im test failing

daehyun99 committed:
Fix: Fix Col2Im test failing

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

daehyun99 commented:
@mvafin 
Oh, I missed that. My apologies.

There was a mistake while applying the changes from this comment:

https://github.com/openvinotoolkit/openvino/pull/33386#issuecomment-3976864560

I fixed them. And I also fixed the error related to xfail_issue_99954.

When you have a moment, could you please review it?

Thanks.

mvafin commented on src/frontends/onnx/frontend/src/op/col2im.cpp:
```diff
@@ -0,0 +1,67 @@
+    // 3. return Col2Im
+    return {
+        std::make_shared<ov::op::v15::Col2Im>(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end)
+            ->outputs()};
```
Other operators don't use `->outputs()`. You better use one or another. You can do this:

```suggestion
    return {
        std::make_shared<ov::op::v15::Col2Im>(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end)};
```

daehyun99 replied:
I applied.

mvafin commented on src/frontends/onnx/tests/tests_python/test_backend.py:
```diff
@@ -381,10 +381,6 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
     (
         xfail_issue_99952,
         "OnnxBackendNodeModelTest.test_col2im_5d_cpu",
```
What error happens when Col2Im is used for 5 dimensional input?

mvafin replied:
Okay, I see that openvino Col2Im doesn't support this case. Lets leave it unsupported for now.

daehyun99 commented on src/frontends/onnx/tests/tests_python/test_backend.py:
```diff
@@ -1,6 +1,6 @@
-# -*- coding: utf-8 -*-
 # Copyright (C) 2018-2026 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
+# -*- coding: utf-8 -*-
```
All Copyright test are passed.

Due to the force-push, the previous test failure logs are no longer available, but as far as I remember, placing it on the first or second line caused an error.


- https://github.com/openvinotoolkit/openvino/actions/runs/24834755480/job/72695256586?pr=33386

daehyun99 commented on src/frontends/onnx/frontend/src/op/col2im.cpp:
```diff
@@ -0,0 +1,65 @@
+    // 2. get attributes
+    const size_t spatial_rank = 2;
+
+    std::vector<size_t> default_attr_vals(spatial_rank, 1);
+    auto dilations = node.get_attribute_value<std::vector<size_t>>("dilations", default_attr_vals);
+    auto strides = node.get_attribute_value<std::vector<size_t>>("strides", default_attr_vals);
+
+    std::vector<size_t> default_pads(spatial_rank * 2, 0);
+    auto pads = node.get_attribute_value<std::vector<size_t>>("pads", default_pads);
+    std::vector<size_t> pads_begin;
+    std::vector<size_t> pads_end;
+
+    // ov::op::v15::Col2Im only supports 2D spatial dimensions
+    CHECK_VALID_NODE(node,
+                     dilations.size() == spatial_rank,
+                     "Col2Im 'dilations' attribute must have size [2]. Got: ",
+                     dilations.size());
+    CHECK_VALID_NODE(node,
+                     strides.size() == spatial_rank,
+                     "Col2Im 'strides' attribute must have size [2]. Got: ",
+                     strides.size());
+    CHECK_VALID_NODE(node,
+                     pads.size() == spatial_rank * 2,
+                     "Col2Im 'pads' attribute must have size [4]. Got: ",
+                     pads.size());
+
```
It is indeed possible to set attribute values to negative numbers in ONNX.

However, after referring to the implementations of other operators, I don’t think this is a good pattern.

By this logic, we would need to implement negative-value handling for attributes in every ONNX frontend operator, but in practice, this is not currently implemented.

daehyun99 committed:
Fix: Apply Review

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

mvafin commented on src/frontends/onnx/tests/tests_python/test_backend.py:
```diff
@@ -1,6 +1,6 @@
-# -*- coding: utf-8 -*-
 # Copyright (C) 2018-2026 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
+# -*- coding: utf-8 -*-
```
```suggestion
# -*- coding: utf-8 -*-
# Copyright (C) 2018-2026 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

mvafin replied:
Copyright check was fixed to allow this. But even if it fails, it is not required, so please follow PEP requirements

daehyun99 replied:
Oh, Okay. Thanks.

mvafin commented on src/frontends/onnx/tests/__init__.py:
```diff
@@ -1,6 +1,6 @@
-# -*- coding: utf-8 -*-
 # Copyright (C) 2018-2026 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
+# -*- coding: utf-8 -*-
```
```suggestion
# -*- coding: utf-8 -*-
# Copyright (C) 2018-2026 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

daehyun99 committed:
Update src/frontends/onnx/tests/tests_python/test_backend.py

Co-authored-by: Maxim Vafin <maxim.vafin@intel.com>

daehyun99 committed:
Update src/frontends/onnx/tests/__init__.py

Co-authored-by: Maxim Vafin <maxim.vafin@intel.com>

daehyun99 committed:
Fix clang-format

mvafin commented:
@daehyun99 please fix code style

daehyun99 committed:
Fix clang-format

daehyun99 commented:
@mvafin Sorry, I fixed them

mvafin commented:
@daehyun99 your changes cannot be built

daehyun99 committed:
Fix build error

mvafin commented:
build_jenkins

daehyun99 commented:
@mvafin Should I take action to fix the failing jenkins tests?

daehyun99 committed:
Merge branch 'master' into ENH/30144-1

mvafin commented:
build_jenkins
